src/renderer/router/index.js

```diff
@@ -14,9 +14,50 @@ import Playlist from '../views/Playlist/Playlist.vue'
 import Channel from '../views/Channel/Channel.vue'
 import Watch from '../views/Watch/Watch.vue'
 
-Vue.use(Router)
+function isQueryEmpty(query) {
+  return (
+    typeof query === 'undefined' ||
+    query === null ||
+    Object.keys(query).length === 0
+  )
+}
 
-const router = new Router({
+class CustomRouter extends Router {
+  push(location) {
+    // only navigates if the location is not identical to the current location
+
+    if (typeof location === 'string') {
+      if (location.includes('?')) {
+        const urlParts = location.split('?')
+
+        if (
+          urlParts[0] !== router.currentRoute.path ||
+          new URLSearchParams(urlParts[1]) !==
+            new URLSearchParams(router.currentRoute.query)
+        ) {
+          return super.push(location)
+        }
+      } else if (
+        location !== router.currentRoute.path &&
+        isQueryEmpty(router.currentRoute.query)
+      ) {
+        return super.push(location)
+      }
+    } else {
+      if (
+        location.path !== router.currentRoute.path &&
+        (isQueryEmpty(location.query) ||
+          location.query !== router.currentRoute.query)
+      ) {
+        return super.push(location)
+      }
+    }
```

Comment:
```suggestion
    const currentQueryUSP = new URLSearchParams(router.currentRoute.query)
    let newPath = ''
    let newQueryUSP = new URLSearchParams()

    if (typeof location === 'string') {
      if (location.includes('?')) {
        const urlParts = location.split('?')
        newPath = urlParts[0]
        newQueryUSP = new URLSearchParams(urlParts[1])
      } else {
        newPath = location
        // newQueryUSP already empty
      }
    } else {
      newPath = location.path
      newQueryUSP = new URLSearchParams(location.query)
    }

    const pathsAreDiff = router.currentRoute.path !== newPath
    // Comparing `URLSearchParams` objects directly will always be different
    const queriesAreDiff = newQueryUSP.toString() !== currentQueryUSP.toString()

    if (pathsAreDiff || queriesAreDiff) {
      return super.push(location)
    }
```

Reply:
Irrelevant now but I think
```js
location.path !== router.currentRoute.path &&
        (isQueryEmpty(location.query) ||
          location.query !== router.currentRoute.query)
```
is a wrong conditional statement
Upper conditional statements are like (new path != cur path **or** new query != cur query) (I think)
This one is (new path != cur path **and** new query != cur query) ?